increased dynamic test suite

BatiGencho opened the pull request.

BatiGencho committed:
extended int.tests suite

BatiGencho committed:
extended int.tests suite

arminsabouri commented on Makefile:
```diff
@@ -354,15 +354,15 @@ pr:
 
 start-btc-server-2:
 	cd ./bin/btc-server && \
-	cargo run --bin btc-server -- --btc-network "${BITCOIND_NETWORK}" --identifier 1 --address 0.0.0.0:8081 --db "./db2" --min-signers 2 --max-signers 2 --toml ./config.toml  --fee-rate-diff-percentage 30 --bitcoind-url "${BITCOIND_URL}" --bitcoind-user "${BITCOIND_USER}" --bitcoind-pass "${BITCOIND_PWD}" --jwt-secret "${NODE_2_DIR}/jwt.hex" --fall-back-fee-rate-sat-per-vbyte 5
```
Why change this?

arminsabouri commented on bin/test-suite/Cargo.toml:
```diff
@@ -35,11 +35,13 @@ reth-btc-wallet = { path = "../../crates/btc-wallet" }
 tempfile.workspace = true
 tracing.workspace = true
 reth-tracing.workspace = true
+reth-ecies.workspace = true
```
do we need ecies as a dependency?

BatiGencho replied:
yes, this is needed as we need to derive the peerid from the pub key which reth needs when we are connecting the peers. There is a method inside of that crate that does this for us.

arminsabouri commented on bin/test-suite/src/mint_contract_abi.rs:
```diff
@@ -7,12 +7,12 @@ pub use mint_contract::*;
     const _: () = {
         ::core::include_bytes!(
-            "/Users/armins/Dev/botanix/macbeth/bin/test-suite/mint_contract_abi.json",
+            "/home/evgeni/Documents/BOTANIX/Macbeth/bin/test-suite/mint_contract_abi.json",
```
Lets change this to `../mint_contract_abi.json`

BatiGencho replied:
done

arminsabouri commented on bin/test-suite/src/suite/consensus/frost/poa_node.rs:
```diff
@@ -284,66 +307,122 @@ pub fn testnet_custom_chain() -> Arc<ChainSpec> {
+        let json = serde_json::to_string_pretty(&rendered_json).unwrap();
+        println!("Rendered botanix testnet configuration {:?}", json);
+
+        let botanix_genesis_filepath = Path::new("./").join("botanix_tesnet.json");
```
Typo
```suggestion
        let botanix_genesis_filepath = Path::new("./").join("botanix_testnet.json");
```

BatiGencho replied:
done

arminsabouri commented on bin/test-suite/src/suite/consensus/frost/test_signing.rs:
```diff
@@ -0,0 +1,247 @@
+        }))
+        .await
+        .map_err(Error::Request)?;
+    */
```
Any reason to comment this out?

arminsabouri replied:
I think this is the most important part of this test

BatiGencho replied:
The signing part fails in the last step with:

 Failed to broadcast tx: JSON-RPC error: RPC error response: RpcError { code: -25, message: "bad-txns-inputs-missingorspent", data: None }
[2024-04-05T12:21:18Z ERROR btc_server::server] INTERNAL ERROR: Failed to finalize signing: Failed to broadcast tx: JSON-RPC error: RPC error response: RpcError { code: -25, message: "bad-txns-inputs-missingorspent", data: None }

arminsabouri replied:
This ussually means the signers signed for utxos that don't exist. I can help with this

arminsabouri commented on bin/test-suite/src/suite/consensus/frost/test_signing.rs:
```diff
@@ -0,0 +1,247 @@
+        }))
+        .await
+        .map_err(Error::Request)?;
+    */
```
Samething here

BatiGencho replied:
same as above ,need to investigate a bit further

arminsabouri commented on bin/test-suite/src/suite/consensus/pbft/mod.rs:
```diff
@@ -0,0 +1 @@
+
```
Can we delete this?

BatiGencho replied:
done, i added a .gitkeep file to remind us that pbft is to be added soon there

arminsabouri commented on output.txt:
```diff
@@ -0,0 +1,484 @@
+cd ./bin/test-suite && \
```
Could delete

BatiGencho replied:
done

arminsabouri reviewed (changes requested):
Great work. Just noticed a few things left in that could get deleted

arminsabouri commented on bin/test-suite/src/suite/consensus/frost/test_signing.rs:
```diff
@@ -0,0 +1,247 @@
+    for _ in 0..suite.global_context.instances {
+        let eth_address = ethers::core::types::Address::random();
+        eth_addresses.push(eth_address);
+    }
```
We should have a hardcoded vec of 2. With two eth addresses here. As there should only be 2 inputs being spent here

arminsabouri commented on bin/test-suite/src/suite/consensus/frost/test_signing.rs:
```diff
@@ -0,0 +1,247 @@
+
+    // get the aggregate pk from any of the clients
+    // Here we are signing for a two inputs that are tweaked differently
+    let mut btc_addresses: Vec<Address> = vec![];
```
Same thing here.

stevenroose commented:
Just a general comment towards the future, when doing a lot of formatting changes, it would be nice if they could be separated from actual code changes in a separate commit or so.

arminsabouri commented on bin/btc-server/src/dkg.rs:
```diff
@@ -86,6 +86,7 @@ impl App {
                 // If we have a max_signers round2 packages we can generate and save the key package
                 let round2_packages = self.db.get_round2_dkg_packages()?;
                 if round2_packages.len() as u16 == self.max_signers - 1 {
+                    // ?
```
Did this get resolved?

BatiGencho replied:
yep, all good here.

arminsabouri commented on bin/btc-server/src/main.rs:
```diff
@@ -134,12 +135,13 @@ impl App {
         let bitcoind_pass = config.bitcoind_pass.clone();
+        let host = config.bitcoind_url.host_str().unwrap_or_default().to_owned();
+        let port = config.bitcoind_url.port_or_known_default().unwrap_or_default().to_owned();
+        let btcd_url = format!("{}:{}", host, port);
```
Nit: btcd is a alternative bitcoin implementation. can we rename to `bitcoind_url`

BatiGencho replied:
changed,

arminsabouri commented on bin/test-suite/src/suite/consensus/frost/btc_server.rs:
```diff
@@ -80,14 +85,21 @@ pub fn clean_db(tasks: &[SpawnedBtcServer]) {
 }
 
-pub fn spawn_n_btc_servers(n: u16, start_port: u16, config: Config) -> Vec<SpawnedBtcServer> {
+pub fn spawn_n_btc_servers(
+    global_context: Arc<GlobalContext>,
+    start_port: u16,
+) -> Vec<SpawnedBtcServer> {
```
Should # of servers come from a function param. I could see some tests needing a specific set up of nodes. Not a request for a change we could update in the future if need be.

BatiGencho replied:
the # servers is the max number of signers. I have removed the instance param.

arminsabouri commented on bin/test-suite/src/suite/consensus/frost/poa_node.rs:
```diff
@@ -284,66 +323,118 @@ pub fn testnet_custom_chain() -> Arc<ChainSpec> {
+            .and_then(|servers| servers.iter().nth(member_index as usize).map(|val| val.port))
+            .unwrap();
+
+        let secp: secp256k1::Secp256k1<secp256k1::All> = secp256k1::Secp256k1::new();
```
Nit: could move this out of the loop

stevenroose replied:
nit: and the type annotation is not needed ;)

BatiGencho replied:
done

arminsabouri commented on bin/test-suite/src/suite/consensus/frost/test_signing.rs:
```diff
@@ -0,0 +1,271 @@
+    assert!(finalized.is_err());
+    let err = finalized.err().unwrap();
+    assert_eq!(err.code(), tonic::Code::Internal);
+    assert!(err.message().contains("bad-txns-inputs-missingorspent"));
```
```suggestion
    // Test should fail here after `psbt.finalize_mut()`. These inputs don't actually exist on the regtest chain so there is nothing to be spent. In the future we can generate some addresses send funds and use those outpoints for this test.
    assert!(err.message().contains("bad-txns-inputs-missingorspent"));
```

BatiGencho replied:
added

arminsabouri reviewed (commented):
Looks great! just a few more nits

stevenroose commented:
I tried to run it, but didn't really work :/ Various errors. Also the hard-coded paths in .env is not ideal.

```
2024-04-09T16:57:41.030045Z ERROR Test suite panicked PanicInfo { payload: Any { .. }, message: Some(called `Result::unwrap()` on an `Err` value: Bitcoind client could not achieve synced status within 60secs. Exiting...
```

Does the suite check if all the configs in `.env` make sense (like if the dirs exist, if the bitcoind is reachable etc) before running the daemons?

arminsabouri committed:
fix(poa): block fetcher should check if imported block exists before executing

arminsabouri committed:
chore(frost): better logging in signing state machine

arminsabouri committed:
chore(btc_server): request logs are too verbose

arminsabouri committed:
fix(frost): use `insert_signing_state()` to update state across signing state machine

the previous function `get_or_insert_signing_state()` only sets signing
state if there is no signing state for a given session id

arminsabouri committed:
fix(frost): signer should update signing state after processing round 2 request

arminsabouri committed:
fix(frost): signer shouldn't check id on signing requests

arminsabouri committed:
chore(btc_server): additional debug statment for adding round 2 signing package

arminsabouri committed:
chore(btc_server|frost): consolidate signing package and signing package request types across signing rounds

arminsabouri committed:
chore(frost): fix typos in error variants

arminsabouri committed:
refactor(engine): validate extra data header should validate everytime

scottmillner committed:
refactor: use timeout when getting new payload
